src/prefect/utilities/logging.py

```diff
@@ -106,9 +110,11 @@ def _write_logs(self) -> None:
         cont = True
         while cont:
             try:
-                while self.pending_length < MAX_BATCH_LOG_LENGTH:
+                # ensuring extra 200 bytes of free batch size for the
+                # log dict properties beside the message with MAX_LOG_LENGTH.
+                while self.pending_length + 200 < MAX_BATCH_LOG_LENGTH:
                     log = self.queue.get_nowait()
-                    self.pending_length += len(log.get("message", ""))
+                    self.pending_length += len(json.dumps(log))
```

Comment:
json.dumps is maybe too expensive here... could be replaced by e.g.
```suggestion
                    self.pending_length += len(log.get("message", "")) + 200
```